contrib/lkt/language/parser.py

```diff
@@ -1473,6 +1462,27 @@ def ref_not_found_error():
             .concat(Self.text).concat(S("` in this scope")),
         )
 
+    @langkit_property(return_type=T.SemanticResult.array)
+    def check_legality():
+        error_type = Var(Entity.property_error_type)
+
+        return If(
+            # Check that we don't reference the error type (PropertyError)
+            # outside of a raise expression
+            # NOTE: This means that we lookup the property error type for every
+            # RefId in the code. That might be too costly, so we might want to
+            # have a first level of textual filter if this ever becomes a
+            # problem.
+            And(Entity.referenced_decl.result_ref == error_type,
+                Not(Entity.parents.any(lambda r: r.is_a(T.RaiseExpr)))),
```

Comment:
Actually your message made me realize that it's worse than that, since currently you can only pass a static string in Langkit to the `PropertyError` constructor.  I'll make the check stricter in `RaiseExpr.check_legality`, which should also take care of the case you're mentioning.